backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java

```diff
@@ -28,16 +33,81 @@ public DefaultUserService(UserDao userDao) {
 
     @Override
     public User getUserById(Long id) {
+        LOG.info("Retrieved user ID {}", id);
         return userDao.findUserById(id);
     }
 
+    /**
+     * Find a unique User with email and username
+     * @param email The email of the user
+     * @param username The username of the user
+     * @return The User in the database
+     */
+    @Override
+    public User getUserByEmailAndUsername(String email, String username) {
+        List<User> userList = userDao.findAll();
+        User userToFind = null;
+
+        for (User user : userList) {
+            if (email.equalsIgnoreCase(user.getEmail()) && username.equalsIgnoreCase(user.getUsername())) {
```

Comment:
Shouldn't this be an OR since we want both unique emails and unique usernames. In this case, if only the email matches, it returns null and vice-versa.

Reply:
Right! I'll make the changes.